### DISCUSSION TITLE
Creating custom expression type

### DISCUSSION THREAD

**Discussion by Gaby-Gabou** [MilesCranmer/PySR#731] [Q&A]
Hi, I'm trying to create a new custom expression type to see if Symbolic regression works with String or Symbol. To create it I used the example of ParametricExpression. However I get an error saying that AbstractExpression is not defined and I don't know why I get this error. Here's the Julia code : 

```
mutable struct StringNode<:SymbolicRegression.AbstractExpressionNode{String}
    val::String
    degree::UInt8


    op::UInt8
    l::StringNode
    r::StringNode

end

struct StringExpr{
    String,
    N<:StringNode,
    D<:NamedTuple{(:operators, :variable_names, :parameters, :parameter_names)},
} <: SymbolicRegression.AbstractExpression{String,N}
    tree::N
    metadata::Metadata{D}

    function ParametricExpression(tree::StringNode, metadata::Metadata)
        return new{eltype(tree),typeof(tree),typeof(_data(metadata))}(tree, metadata)
    end
end
```

Thanks in advance for the help

**Comment by MilesCranmer** [comment]
What version of SymbolicRegression.jl are you using? 1.0.0 is in beta so you need to install from the GitHub directly

**Reply by Gaby-Gabou** [reply]
I'm using the 0.24.5 version. So I can try it with the 1.0.0 version and see how it works

**Reply by MilesCranmer** [reply]
Oh yeah the AbstractExpression is only on 1.0.0+

**Reply by Gaby-Gabou** [reply]
Okay so now it works. Thank you very much
